cursorless.nvim/lua/cursorless/cursorless.lua

```diff
@@ -11,114 +11,45 @@ function M.window_get_visible_lines()
   return { vim.fn.line("w0"), vim.fn.line("w$") }
 end
 
--- https://www.reddit.com/r/neovim/comments/p4u4zy/how_to_pass_visual_selection_range_to_lua_function/
--- https://neovim.io/doc/user/api.html#nvim_win_get_cursor()
+-- Get the coordinates of the current selection
+-- To manually test follow these steps:
+-- 1. In command mode :vmap <c-a> <Cmd>lua print(vim.inspect(require('cursorless').buffer_get_selection()))<Cr>
+-- 2. type "hello" on the first line and "world" on the second line
+-- 3. Enter visual mode and select "hello" on the first line and continue selection with "air"
+--    on the second line.
+-- 4. Hit ctrl+b to show the selection: {1, 1, 2, 5, false}
+-- 5. Hit 'o' to swap the cursor position and hit ctrl+b again: {1, 1, 2, 5, true}
```

Comment:
ctrl+a

Reply:
is that normal that by switching the cursor position we get the same result? Shouldn't we get it in reversed order?

Reply:
> is that normal that by switching the cursor position we get the same result? Shouldn't we get it in reversed order?

It is the way it behaves atm ya, not a typo in the comment. I kept your logic for that from the original code the same. It's the true/false that indicates the order is reversed, and I guess the typescript special cases when it's reversed. It could be the typescript handling could be simplified/changed too but I'll have to look again. 

The fact the tests all pass to me implies that it being the way it is still correct.